src/openlrr/game/world/Roof.cpp

```diff
@@ -22,30 +24,304 @@ LegoRR::Roof_Globs & LegoRR::roofGlobs = *(LegoRR::Roof_Globs*)0x00553180;
 #pragma region Functions
 
 // <LegoRR.exe @00463060>
-//bool32 __cdecl LegoRR::Roof_Initialise(Gods98::Container* contRoot, sint32 width, sint32 height);
+bool32 __cdecl LegoRR::Roof_Initialise(Gods98::Container* contRoot, sint32 width, sint32 height)
+{
+	// Make mesh from base container, check if its successful afterwards
+	roofGlobs.contMesh = Gods98::Container_MakeMesh2(contRoot, Gods98::Container_MeshType::Immediate);
+
+	if (roofGlobs.contMesh != nullptr) {
+		// clear textures and set up counter for looping through grid
+		roofGlobs.texture = nullptr;
+		uint32 count = height * width;
+		roofGlobs.hidden = 0;
+		roofGlobs.grid = (RoofBlock*)Gods98::Mem_Alloc(count * sizeof(RoofBlock));
+		// if we were able to create the grid start looping through it
+		if (roofGlobs.grid != nullptr) {
+			if (count != 0) {
```

Comment:
The `count != 0` check is a common part of when ghidra uses a `do`/`while` loop instead of a `for` loop. This check can be removed, since the same behaviour is handled by the for loop itself.